input/pagecontent/algorithms.md

```diff
@@ -1,257 +1,189 @@
 
-```
+##### De-Identifying Dates
 
-**A08 after replacing values with fixed values:**
+Dates are a common quasi-identifier and must be handled carefully to prevent re-identification. Common techniques include:
 
-```hl7
-MSH\|^\~\\&\|PROACCESS5\|DHIN\|BIOSENSE\|CDC01\|20080808290000\|\|ADT^A08\|111122223333\|P\|2.5\|\|
+* **Date Shifting:** All dates for a specific individual are shifted by a random (but consistent) number of days. This preserves the interval between events but obscures the actual dates.
+* **Coarsening:** The precision of the date is reduced. For example, a specific birth date (e.g., "1985-04-12") is replaced with just the year ("1985") or a broader range (e.g., "1985-1990").
+* **Converting to Age:** Dates can be converted into an age at the time of an event. This age can then be generalized (e.g., grouped into 5-year brackets) to further reduce risk.
 
-PID\|123\|123\|123456\|123\|FamilyName^""^GivenName^""^""^""\|\|19900113\|U\|Alias\|U\|""^""^""^^840??^""\|US\|HomePh\|BusPh\|U\|U\|U\|PSEUDO1234\|U\|U\|U\|U\|U\|""
+##### De-Identifying Geographical Locations and Geolocation Data
 
-```
+Geographic data is highly identifying. Techniques for de-identifying it include:
 
-#### Other Considerations
+* **Generalization:** Replace precise addresses with broader regions, such as zip codes, counties, or states. The level of generalization depends on the population density of the area.
+* **Geometric Aggregation:** Combine geographic points into larger polygons, obscuring the exact location of individuals.
+* **Suppression:** Remove location data entirely for individuals in sparsely populated areas where they could be easily identified.
 
-Care must be taken to ensure redacted data remain syntactically correct. A HL7 CDA document that complies with a template has specific rules regarding discrete data elements and required structure. Post processing after de-identification may be required to create a document that still complies with the template. Template designers may need to consider the needs of de-identification in the design of new templates.
+##### De-Identifying Genomic Information
 
-### Fuzzing
+Genomic data is unique and inherently identifying. De-identification is extremely challenging and often focuses on risk mitigation rather than complete anonymization. Key strategies include:
 
-#### Description
+* **Controlled Access:** Restricting access to genomic data to authorized researchers under strict data use agreements.
+* **Data Partitioning:** Splitting genomic data from demographic and clinical data, with access controls on both.
+* **Cryptographic Methods:** Using techniques like homomorphic encryption to allow computation on encrypted genomic data without decrypting it.
 
```

Comment:
Why not the methods of generalization? One can certainly treat the genomic data as a set of codes, for which all the techniques that one can apply to codes can be applied here. Yes much of the data are too specific, so can't be generalied and thus will need to be redacted.